traffic_ops/testing/api/v14/stats_summary_test.go

```diff
@@ -0,0 +1,186 @@
+)
+
+const dateFormat = "2006-01-02"
+
+func TestStatsSummary(t *testing.T) {
+	WithObjs(t, []TCObj{}, func() {
+		testStatsSummaries = []tc.StatsSummary{}
+		latestTime = time.Now().Truncate(time.Second).UTC()
+		CreateTestStatsSummaries(t)
+		GetTestStatsSummaries(t)
+		GetTestStatsSummariesLastUpdated(t)
+	})
+}
+
+func CreateTestStatsSummaries(t *testing.T) {
+	tmpTime := latestTime
+	for _, ss := range testData.StatsSummaries {
+		ss.SummaryTime = tmpTime
+		timeStamp, err := time.Parse(dateFormat, tmpTime.Format(dateFormat))
+		if err != nil {
+			t.Errorf("error getting timestamp at beginning of the day: %v", err)
+		}
+
+		ss.StatDate = timeStamp
+
+		_, _, err = TOSession.CreateSummaryStats(ss)
+		if err != nil {
+			t.Errorf("creating stats_summary %v: %v", ss.StatName, err)
+		}
+
+		tmpTime = tmpTime.AddDate(0, 0, -1)
+
+		testStatsSummaries = append(testStatsSummaries, ss)
+	}
+}
+
+func GetTestStatsSummaries(t *testing.T) {
+	var testCases = []struct {
+		description            string
+		stat                   *string
+		cdn                    *string
+		ds                     *string
+		expectedStatsSummaries []tc.StatsSummary
+	}{
+		{
+			description:            "get all summary stats",
+			expectedStatsSummaries: testStatsSummaries,
+		},
+		{
+			description: "non-existant stat name",
+			stat:        util.StrPtr("bogus"),
+		},
+		{
+			description: "non-existant ds name",
+			ds:          util.StrPtr("bogus"),
+		},
+		{
+			description: "non-existant cdn name",
+			cdn:         util.StrPtr("bogus"),
+		},
+		{
+			description: "get stats summary by stat name",
+			stat:        util.StrPtr("daily_bytesserved"),
+			expectedStatsSummaries: func() []tc.StatsSummary {
+				statsSummaries := []tc.StatsSummary{}
+				for _, ss := range testStatsSummaries {
+					if ss.StatName == "daily_bytesserved" {
+						statsSummaries = append(statsSummaries, ss)
+					}
+				}
+				return statsSummaries
+			}(),
+		},
+		{
+			description:            "get stats summary by cdn name",
+			cdn:                    util.StrPtr("cdn1"),
+			expectedStatsSummaries: testStatsSummaries,
+		},
+		{
+			description:            "get stats summary by ds name",
+			ds:                     util.StrPtr("all"),
+			expectedStatsSummaries: testStatsSummaries,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			tsr, _, err := TOSession.GetSummaryStats(tc.cdn, tc.ds, tc.stat)
+			if err != nil {
+				t.Fatalf("received unexpectant error %v on GET to stats_summary", err)
+			}
+			if len(tc.expectedStatsSummaries) == 0 && len(tsr.Response) != 0 {
+				t.Fatalf("expected to recieve no stats summaries but received %v", len(tsr.Response))
+			}
+			for _, ess := range tc.expectedStatsSummaries {
+				found := false
+				for _, ss := range tsr.Response {
+					if ess == ss {
+						found = true
+						break
+					}
+				}
+				if !found {
+					t.Fatalf("expected to find stat %v in stats summary response", ess.StatName)
```

Comment:
I think if you make this a non-fatal error you can get a more exhaustive list of what's wrong without risking anything due to bad state/data